cmd/vulcan-govulncheck/main.go

```diff
@@ -0,0 +1,208 @@
+		args = append(args, "-db", opt.DB)
+	}
+	if opt.Mode != "" {
+		args = append(args, "-mode", opt.Mode)
+	}
+	if opt.Scan != "" {
+		args = append(args, "-scan", opt.Scan)
+	}
+	if opt.Tags != "" {
+		args = append(args, "-tags", opt.Tags)
+	}
+	if opt.Test != "" {
+		args = append(args, "-test", opt.Test)
+	}
+	if opt.Patterns != "" {
+		args = append(args, opt.Patterns)
+	}
+
+	var stdout, stderr bytes.Buffer
+	cmd := exec.CommandContext(ctx, "govulncheck", args...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	cmd.Dir = dir
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("run command: %w: %q", err, stderr.String())
+	}
+
+	vulns, err := parseSarif(&stdout)
+	if err != nil {
+		return fmt.Errorf("parse sarif: %w", err)
+	}
+
+	state.AddVulnerabilities(vulns...)
+	return nil
+}
+
+func parseSarif(r io.Reader) ([]report.Vulnerability, error) {
+	l, err := sarif.Decode(r)
+	if err != nil {
+		return nil, fmt.Errorf("decode SARIF document: %w", err)
+	}
+
+	var vulns []report.Vulnerability
+	for _, run := range l.Runs {
+		for _, result := range run.Results {
+			vuln, err := parseResult(l, result)
+			if err != nil {
+				return nil, fmt.Errorf("parse result: %w", err)
+			}
+			vulns = append(vulns, vuln)
+		}
+	}
+	return vulns, nil
+}
+
+func parseResult(l sarif.Log, result sarif.Result) (report.Vulnerability, error) {
+	rule, found := l.FindRule(result.RuleID)
+	if !found {
+		return report.Vulnerability{}, fmt.Errorf("unknown rule: %v", result.RuleID)
+	}
+
+	var rscs []report.ResourcesGroup
+	for _, cf := range result.CodeFlows {
+		rsc := report.ResourcesGroup{
+			Name:   cf.Message.Text,
+			Header: []string{"module", "location", "function"},
+		}
+		for _, tf := range cf.ThreadFlows {
+			for _, loc := range tf.Locations {
+				row := make(map[string]string)
+				row["module"] = loc.Module
+				row["location"] = loc.Location.PhysicalLocation.String()
+				row["function"] = loc.Location.Message.Text
+				rsc.Rows = append(rsc.Rows, row)
+			}
+		}
+		rscs = append(rscs, rsc)
+	}
+
+	location := getLocations(result)
+	vuln := report.Vulnerability{
+		Summary:          rule.ShortDescription.Text,
+		Score:            calcScore(result),
+		AffectedResource: location,
+		Fingerprint:      helpers.ComputeFingerprint(result.RuleID, location),
+		CWEID:            937,
+		Description:      rule.FullDescription.Text,
+		Details:          result.Message.Text,
+		Labels:           []string{"issue"},
+		Recommendations: []string{
+			"Visit the linked references for more details.",
+		},
+		References: []string{rule.HelpURI},
+		Resources:  rscs,
+	}
+
+	return vuln, nil
+}
+
+func getLocations(result sarif.Result) string {
+	var locs []string
+	for _, loc := range result.Locations {
+		locs = append(locs, loc.PhysicalLocation.String())
+	}
+	return strings.Join(locs, " ")
+}
+
+func calcScore(result sarif.Result) float32 {
+	if result.Level == "error" {
+		return 8 // high severity
+	} else {
+		return 0 // informational severity
```

Comment:
As `vulcan-report` package is already imported, I may suggest to use [report.SeverityThresholdNone](https://github.com/adevinta/vulcan-report/blob/master/utils.go#L31) in order to return a Info vulnerability score.

Reply:
I like it. It makes the code self-explanatory. Fixed.